Handle scrollend events for select elements

nmoucht opened the pull request:
#### ef277d5f855b6e8178298e2f68e5da6cf6557e39
<pre>
Handle scrollend events for select elements
<a href="https://bugs.webkit.org/show_bug.cgi?id=296977">https://bugs.webkit.org/show_bug.cgi?id=296977</a>
<a href="https://rdar.apple.com/157610355">rdar://157610355</a>

Reviewed by Simon Fraser.

Handle scrollend events for select elements.

* LayoutTests/fast/scrolling/mac/scrollend-event-on-select-element-expected.txt: Added.
* LayoutTests/fast/scrolling/mac/scrollend-event-on-select-element.html: Added.
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::scrollableAreaCanHandleEvent):
(WebCore::EventHandler::handleWheelEventInScrollableArea):
* Source/WebCore/platform/ScrollAnimator.cpp:
(WebCore::ScrollAnimator::handleSteppedScrolling):
* Source/WebCore/rendering/RenderListBox.cpp:
(WebCore::RenderListBox::scrollDidEnd):
* Source/WebCore/rendering/RenderListBox.h:

Canonical link: <a href="https://commits.webkit.org/298402@main">https://commits.webkit.org/298402@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ccdde34df2cd1b45477689d1f95acebec10bfc27

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/115323 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/35017 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/25519 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/121414 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/65910 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/117212 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/35676 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/43589 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/87609 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/42338 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d921d873-c7b6-4b3a-bcb7-b0ea0dd9642e) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/118271 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/28443 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/103520 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/68005 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/27600 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/21642 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/65077 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/97828 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/21756 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/124583 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/42263 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/31646 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/96397 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/42631 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/99709 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/96182 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24489 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/41412 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/19266 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/38201 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/42145 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/47699 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/41656 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/44981 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/43381 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

Ahmad-S792 commented on LayoutTests/fast/scrolling/mac/scrollend-event-on-select-element.html:
```diff
@@ -0,0 +1,135 @@
+        }
+        
+    </style>
+    <script src="../../../resources/js-test-pre.js"></script>
```
`js-test.js` instead of `pre/post`.

nmoucht replied:
Fixed.

smfr commented on Source/WebCore/platform/ScrollAnimator.cpp:
```diff
@@ -182,6 +182,13 @@ bool ScrollAnimator::handleWheelEvent(const PlatformWheelEvent& wheelEvent)
 // "Stepped scrolling" is only used by RenderListBox. It's special in that it has no rubberbanding, and scroll deltas respect Scrollbar::pixelStep().
 bool ScrollAnimator::handleSteppedScrolling(const PlatformWheelEvent& wheelEvent)
 {
+#if PLATFORM(MAC)
```
Should these be `#if ENABLE(KINETIC_SCROLLING)` instead? Is this really macOS only?

nmoucht replied:
Fixed.

nmoucht committed:
Handle scrollend events for select elements
https://bugs.webkit.org/show_bug.cgi?id=296977
rdar://157610355

Reviewed by Simon Fraser.

Handle scrollend events for select elements.

* LayoutTests/fast/scrolling/mac/scrollend-event-on-select-element-expected.txt: Added.
* LayoutTests/fast/scrolling/mac/scrollend-event-on-select-element.html: Added.
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::scrollableAreaCanHandleEvent):
(WebCore::EventHandler::handleWheelEventInScrollableArea):
* Source/WebCore/platform/ScrollAnimator.cpp:
(WebCore::ScrollAnimator::handleSteppedScrolling):
* Source/WebCore/rendering/RenderListBox.cpp:
(WebCore::RenderListBox::scrollDidEnd):
* Source/WebCore/rendering/RenderListBox.h:

Canonical link: https://commits.webkit.org/298402@main
